### PR TITLE
kea: add new package

### PR DESCRIPTION
**Pull request by hbl0307106015**
Kea is an open source DHCPv4/DHCPv6 server being developed by
Internet Systems Consortium. Kea is a high-performance, extensible
DHCP server engine that is designed to be easily modified and extended
with hooks libraries.

Kea is free open source, and we welcome community engagement, via the
Kea-users mailing list, this wiki, and our Github
repository. There is a small core team of dedicated software engineers
developing it and we need your contributions and support
contracts to support them.

DHCP Standardization efforts: The lead developer on KEA is
co-chair of the Dynamic Host Configuration working group in the
IETF. We are committed to providing a standards-compliant
implementation and are closely tracking developments in this working
group and evaluating them for inclusion in KEA.

wiki : http://kea.isc.org/wiki
official : https://www.isc.org/kea/
github : https://github.com/isc-projects/kea

Signed-off-by: BangLang Huang <banglang.huang@foxmail.com>

### CODE REVIEW

**Comment by hbl0307106015** [comment]
This pull request need the package PR#4930  and  PR#4932

**Comment by diizzyy** [comment]
Out of curiosity, is there are reason why you need this? It's kinda beefy and no that great for embedded platforms?

**Comment by hbl0307106015** [comment]
@diizzyy  The description from https://www.isc.org/kea/ can answer your question.

## Why choose Kea?

**Add and change subnets and pools without restarting Kea**.
Store leases and host reservations in a structured database rather than a text file.
**Extend and customize Kea through a Hooks api.**
Kea has comprehensive Developer and Administrator documentation.
ISC offers commercial 7 x 24 support for Kea, as well as consulting and contract development to assist in implementing Kea, including migration from ISC DHCP.
Kea runs on Linux, BSD, and MacOS, like ISC DHCP.   The Kea distribution does not yet include a DHCP client or relay, but because both are standards-based, the ISC DHCP client works fine with the Kea DHCP server.  Kea is under active development.

In 2017 we are adding a remote management capability.  This work is generously sponsored by Mozilla, through a Mozilla Open Source award.

**Comment by diizzyy** [comment]
No it doesn't really...

**Comment by diizzyy** [comment]
@hbl0307106015 
Please update to the latest release if you want it in

**Comment by hbl0307106015** [comment]
@diizzyy  Ok, thanks

**Comment by hbl0307106015** [comment]
It's done, here is the newer commit, please check it again, thanks! @diizzyy @rosysong

**Comment by hbl0307106015** [comment]
@diizzyy  Hi, what should I do next ? Is there any possible to merge it ?

**Comment by diizzyy** [comment]
@hbl0307106015 
Sorry missed it...
kea (host-package) doesn't compile

```
...
checking log4cplus/logger.h usability... no
checking log4cplus/logger.h presence... no
checking for log4cplus/logger.h... no
configure: error: Missing required header files.
configure: error: Missing required header files.
Makefile:172: recipe for target '/home/build/build_dir/build_dir/hostpkg/kea-381-bad-config-crashes-ca-or-d2-servers-even-with-c_base/.configured' failed
make[2]: *** [/home/build/build_dir/build_dir/hostpkg/kea-381-bad-config-crashes-ca-or-d2-servers-even-with-c_base/.configured] Error 1
make[2]: Leaving directory '/home/build/openwrt_packages/net/kea'
```

**Comment by hbl0307106015** [comment]
@diizzyy Yes, the kea package need log4cpuls. https://github.com/openwrt/packages/pull/4932

File: net/kea/Makefile [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=kea
+PKG_VERSION:=460-update-mysql-database-schema-for-dhcpv6-config-backend--port_base
+PKG_RELEASE:=1
+PKG_MAINTAINER:=BangLang Huang<banglang.huang@foxmail.com> \
+  Rosy Song<rosysong@rosinson.com>
```
Comment (neheb) [inline net/kea/Makefile]: Please match identation.

File: net/kea/Makefile [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+PKG_BUILD_DEPENDS:=kea/host
+# real dir name in released tar.gz file
+RELEASE_NAME:=460-update-mysql-database-schema-for-dhcpv6-config-backend-port_base
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(RELEASE_NAME)
+HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/$(PKG_NAME)-$(RELEASE_NAME)
+
+PKG_CAT:=zcat
+PKG_SOURCE:=$(PKG_VERSION)
+PKG_SOURCE_URL:=https://codeload.github.com/isc-projects/kea/tar.gz/
```
Comment (neheb) [inline net/kea/Makefile]: upstream has moved to GitLab

File: net/kea/Makefile [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+# Copyright (C) 2019 Rosy Song <rosysong@rosinson.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=kea
+PKG_VERSION:=460-update-mysql-database-schema-for-dhcpv6-config-backend--port_base
```
Comment (neheb) [inline net/kea/Makefile]: This is not only a beta version but also a ridiculous name for it. The latest stable version is 1.5.0.

File: net/kea/Makefile [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+HOST_BUILD_DIR:=$(BUILD_DIR_HOST)/$(PKG_NAME)-$(RELEASE_NAME)
+
+PKG_CAT:=zcat
+PKG_SOURCE:=$(PKG_VERSION)
+PKG_SOURCE_URL:=https://codeload.github.com/isc-projects/kea/tar.gz/
+PKG_HASH:=0951e513e8907a58ebde61a0a7005a71d22c6bd4270390f0dbb050f28847a054
+
+PKG_INSTALL:=1
+PKG_LICENSE:=MPL-2.0
+PKG_FIXUP:=autoreconf
```
Comment (neheb) [inline net/kea/Makefile]: As this is using autotools, might want to add PKG_BUILD_PARALLEL for faster compilation.

File: net/kea/Makefile [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+CONFIGURE_VARS += \
+	cross_compiling="yes"
+
+TARGET_CXXFLAGS += \
+		-std=c++11 \
+		-fdata-sections \
+		-ffunction-sections
+
+TARGET_LDFLAGS += \
+		-Wl,--gc-sections
```
Comment (neheb) [inline net/kea/Makefile]: Adding --as-needed at the end can result in a smaller size.

File: net/kea/Makefile [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+CONFIGURE_ARGS += \
+	--with-log4cplus="$(STAGING_DIR)/usr/include" \
+	$(if $(CONFIG_PACKAGE_kea-perfdhcp),--enable-perfdhcp,)
+
+CONFIGURE_VARS += \
+	cross_compiling="yes"
+
+TARGET_CXXFLAGS += \
+		-std=c++11 \
```
Comment (neheb) [inline net/kea/Makefile]: This is no longer needed with GCC7 AFAIK.

**Comment by hbl0307106015** [comment]
@neheb @diizzyy  It's done , please check it again.

File: net/kea/Makefile [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+CONFIGURE_VARS += \
+	cross_compiling="yes"
+
+HOST_CONFIGURE_ARGS += \
+	--enable-static \
+	--enable-static-link \
+	--with-log4cplus="$(STAGING_DIR_HOSTPKG)" \
+
+HOST_CXXFLAGS += \
```
Comment (neheb) [inline net/kea/Makefile]: Not sure about this. The TARGET_FLAGS make sense but not so much the HOST. I don't think space is a concern on the host.

File: net/kea/Makefile [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+PKG_NAME:=kea
+PKG_VERSION:=1.5.0
+PKG_RELEASE:=1
+PKG_MAINTAINER:=BangLang Huang<banglang.huang@foxmail.com>, Rosy Song<rosysong@rosinson.com>
+PKG_BUILD_DEPENDS:=log4cplus kea/host
+HOST_BUILD_DEPENDS:=log4cplus/host boost/host
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=http://ftp.isc.org/isc/kea/$(PKG_VERSION)/
```
Comment (neheb) [inline net/kea/Makefile]: From a quick test, HTTPS is possible here.

Comment (hbl0307106015) [reply net/kea/Makefile]: done

File: net/kea/Makefile (hunk 7 above)
Comment (hbl0307106015) [reply net/kea/Makefile]: Well, removed it.

File: net/kea/Makefile [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+PKG_NAME:=kea
+PKG_VERSION:=1.5.0
+PKG_RELEASE:=1
+PKG_MAINTAINER:=BangLang Huang<banglang.huang@foxmail.com>, Rosy Song<rosysong@rosinson.com>
+PKG_BUILD_DEPENDS:=log4cplus kea/host
+HOST_BUILD_DEPENDS:=log4cplus/host boost/host
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://ftp.isc.org/isc/kea/$(PKG_VERSION)/ \
+	https://codeload.github.com/isc-projects/kea/tar.gz/Kea-$(PKG_VERSION)?
```
[the remark this answers is not in the document]
Comment (hbl0307106015) [reply net/kea/Makefile]: Well, removed, thanks!  But note that the official download site occured errors yesterday, thus I added another choice on its github repo.

**Comment by neheb** [comment]
Please rebase against current master so that CircleCI can build this.

**Comment by hbl0307106015** [comment]
@neheb  Thanks, it's done.

**Comment by neheb** [comment]
Great. Merging.

**Pull request merged**